src/net/client/request.rs

```diff
@@ -33,18 +33,21 @@ pub trait ComposeRequest: Debug + Send + Sync {
     ) -> Result<(), CopyRecordsError>;
 
     /// Create a message that captures the recorded changes.
-    fn to_message(&self) -> Message<Vec<u8>>;
+    fn to_message(&self) -> Result<Message<Vec<u8>>, Error>;
 
     /// Create a message that captures the recorded changes and convert to
     /// a Vec.
-    fn to_vec(&self) -> Vec<u8>;
+    fn to_vec(&self) -> Result<Vec<u8>, Error>;
 
     /// Return a reference to a mutable Header to record changes to the header.
     fn header_mut(&mut self) -> &mut Header;
 
     /// Set the UDP payload size.
     fn set_udp_payload_size(&mut self, value: u16);
 
+    /// Set the DNSSEC OK flag.
+    fn set_dnssec_ok(&mut self, value: bool);
```

Comment:
Should this maybe be more generic, e.g. `fn set_edns_flag(FlagEnum)` ?

Reply:
As far is I know, that doesn't exist in OptBuilder. As far as I know the only flag in existence is DO.